packages/opentelemetry-exporter/src/index.ts

```diff
@@ -0,0 +1,211 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+import { ExportResult, ExportResultCode } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/sdk-trace-base';
+const { backendConnector: instanaBackendConnector, environment: instanaEnvironment } = require('@instana/serverless');
+// TODO: add these constants to the constant module
+const instanaEndpointUrlEnvVar = 'INSTANA_ENDPOINT_URL';
+const instanaAgentKeyEnvVar = 'INSTANA_AGENT_KEY';
+
+/**
+ * The exporter is based on the parser from the Agent:
+ * https://github.com/instana/agent/blob/51da097e99e9d05ac5c0b5b7dd9275a12c20685d/agent-optional-parent/agent-open-telemetry/src/main/java/com/instana/agent/main/impl/otel/OpenTelemetryTraceEndpoint.java#L69
+ */
+
+/** Span Kind
+Default value. Indicates that the span is used internally.
+INTERNAL = 0,
+
+Indicates that the span covers server-side handling of an RPC or other
+remote request.
+SERVER = 1,
+
+Indicates that the span covers the client-side wrapper around an RPC or
+other remote request.
+CLIENT = 2,
+
+Indicates that the span describes producer sending a message to a
+broker. Unlike client and server, there is no direct critical path latency
+relationship between producer and consumer spans.
+PRODUCER = 3,
+
+Indicates that the span describes consumer receiving a message from a
+broker. Unlike client and server, there is no direct critical path latency
+relationship between producer and consumer spans.
+CONSUMER = 4
+*/
+
+const SPAN_KIND_INTERNAL = 0;
+const SPAN_KIND_SERVER = 1;
+const SPAN_KIND_CLIENT = 2;
+const SPAN_KIND_PRODUCER = 3;
+const SPAN_KIND_CONSUMER = 4;
+
+const SERVER_KIND = 'server';
+const CLIENT_KIND = 'client';
+const PRODUCER_KIND = 'producer';
+const CONSUMER_KIND = 'consumer';
+const INTERNAL_KIND = 'internal';
+
+/**
+ * Status Code
+ *
+ * Default 
+ * UNSET = 0,
+ * The operation has been validated by an Application developer or
+ * Operator to have completed successfully.
+ * OK = 1,
+ * The operation contains an error.
+ * ERROR = 2
+ */
+const STATUS_UNSET = 0;
+const STATUS_OK = 1;
+const STATUS_ERROR = 2;
+
+const statusAsString = {
+  2: 'ERROR',
+  1: 'OK',
+  0: 'UNSET'
+};
+
+export class InstanaExporter implements SpanExporter {
+
+  constructor(agentKey?: string, endpointUrl?: string) {
+
+    // If endpoint URL and agent key are not provided, we don't set them in an attempt to use env vars, if they were set
+    if (endpointUrl) {
+      process.env[instanaEndpointUrlEnvVar] = endpointUrl;
+    }
+
+    if (agentKey) {
+      process.env[instanaAgentKeyEnvVar] = agentKey;
+    }
+
+    instanaEnvironment.validate();
+    instanaBackendConnector.init();
```

Comment:
I think we should provide an `identityProvider` here to make sure `span.f` and the x-instana-host header are set correctly in backend_connector.